src/pages/Vaults.tsx

```diff
@@ -137,6 +138,30 @@ export function Vaults() {
     error,
   } = useSWR<AxiosResponse, AxiosError>(VAULTS_DASHBOARD_QUERY, subQueryFetcher);
 
+  const vaultDataResponse: VaultsDashboardResponse = vaultsData?.data.data;
+  const [vaultsDataAppended, setVaultsDataAppended] = useState<VaultsDashboardResponse | null>(null);
+
+  const totalVaultsCount = vaultDataResponse?.vaults.totalCount || 1;
+  const pageCount = Math.ceil(totalVaultsCount / 100) - 1;
+  const { data: vaultsNextPages } = useSWR<AxiosResponse, AxiosError>(
+    pageCount ? OPEN_VAULTS_NEXT_PAGES_QUERY(pageCount) : null,
+    subQueryFetcher,
+  );
+  useEffect(() => {
+    if ((pageCount !== 0 && !vaultsNextPages) || !vaultDataResponse) return;
```

Comment:
I think you can get the same result without using `useEffect` and `useState`. It doesn't seems appropriate to use these hooks given we're using `useSwr`. 

I did this and it was working:
```
  const vaultDataResponse: VaultsDashboardResponse = vaultsData?.data.data;
  let vaultsDataAppended: VaultsDashboardResponse | null = null;

  const totalVaultsCount = vaultDataResponse?.vaults.totalCount || 1;
  const pageCount = Math.ceil(totalVaultsCount / 100) - 1;
  const { data: vaultsNextPages } = useSWR<AxiosResponse, AxiosError>(
    pageCount ? OPEN_VAULTS_NEXT_PAGES_QUERY(pageCount) : null,
    subQueryFetcher,
  );

  if (vaultsNextPages) {
    if ((pageCount !== 0 && !vaultsNextPages) || !vaultDataResponse) return;
    const vaultPages: {
      [key: string]: {
        nodes: Array<VaultsNode>;
      };
    } = vaultsNextPages?.data.data || {};
    const nextVaults = Object.values(vaultPages).flatMap((openVaultsPage) => openVaultsPage.nodes);
    vaultsDataAppended = {
      ...vaultDataResponse,
      vaults: { ...vaultDataResponse.vaults, nodes: [...vaultDataResponse.vaults.nodes, ...nextVaults] },
    };
  }
```